site/src/pages/release-notes/1.32.0.mdx

```diff
@@ -0,0 +1,183 @@
+        return new ServerStopException();
+      })
+      .build();
+  Server.builder()
+        .gracefulShutdown(gracefulShutdown);
+  ```
+- **DNS Channel Strategy**: You can now set `datagramChannelStrategy` which provides an option to create a
+  channel per-resolution for DNS queries. #6122 #6127
+  ```java
+  DnsAddressEndpointGroup
+    .builder("foo.com")
+    .port(8080)
+    .datagramChannelStrategy(
+      DnsNameResolverChannelStrategy.ChannelPerResolution) // 👈👈👈
+    .build();
+  ```
+- **Map Injection for an Annotated Service**: The annotated service now supports the `Map<String, Object>`
+  parameter. #6058 #6072
+  ```java
+  @Get("/dogma/repos")
+  public String repos(@Param Map<String, Object> repoOptions) {...}
+  ```
+- **Eureka InstanceInfo**: You can now retrieve a Eureka <type://InstanceInfo> from an <type://Endpoint>
+  via <type://InstanceInfo#instanceInfo(Endpoint)>. #6056 #6069
+  ```java
+  EurekaEndpointGroup eurekaEndpointGroup = ...;
+  eurekaEndpointGroup.endpoints().forEach(endpoint -> {
+    InstanceInfo instanceInfo = InstanceInfo.instanceInfo(endpoint); // 👈👈👈
+  ```
+- **Client Preprocessor**: You can now use a client <type://Preprocessor> to customize certain properties. #6057
+  ```java
+  HttpPreprocessor preprocessor = (delegate, ctx, req) -> {
+    ctx.setEndpointGroup(Endpoint.of("overriding-host"));
+    return delegate.execute(ctx, req);
+  };
+  WebClient client =
+    WebClient.builder()
+             .preprocessor(preprocessor) // 👈👈👈
+             .build();
+  ```
+- **Backoff Builder**: You can now easily create a <type://Backoff> using builder. #5483 #5488
+  ```java
+  Backoff backoff2 =
+    Backoff.builderForExponential()
+           .initialDelayMillis(100)
+           .maxDelayMillis(10000)
+           .multiplier(2.0)
+           .build();
+  ```
+
+## 📈 Improvements
+
+- Armeria now automatically registers the metrics of `PooledByteBufAllocator.DEFAULT`. #2633 #5916
+- Server metrics now have a port tag for better monitoring and troubleshooting. #6089 #6116
+- You can now see the remoteAddress in the error message when a <type://SessionProtocolNegotiationException>
+  is raised. #6113
+- You can now use the default <type://HealthCheckUpdateHandler> via <type://HealthCheckUpdateHandler#of()> #6002
+
+## 🛠️ Bug fixes
+
+- A cookie is correctly overwritten when a cookie with the same name, domain, and path is set. #5870 #6126
+- <type://CorsPolicyBuilder#build()> is exposed publicly again. #6121
+- Connections do not hold references on automatically aborted requests, resulting in less memory pressure. #6108
+- <type://CircuitBreakerRuleBuilder#onTimeException()> and <Type://RetryRuleBuilder#onTimeoutException()>`
+  now work when a connection or proxy establishment is timed out. #6107
+- DNS refresh query is executed by the correct executor. #5891 #6092
+- You can now choose whether to use the `json_name` field option when transcoding HTTP/JSON to gRPC messages. #5890 #6082
```

Comment:
Should we mention this in the breaking changes section as well?

Reply:
Forgot about that. Updated it. 😉